### ISSUE TITLE
Add cable type 'other' when connecting interfaces

### ISSUE THREAD

**Issue by cyr0nk0r** [netbox-community/netbox#10175]
### NetBox version

v3.3.0

### Feature type

Change to existing functionality

### Proposed functionality

When connecting 2 interfaces together (A side, B side) you select the cable type from a drop down list.
There are quite a few different types of cables, but sometimes the cable you're using doesn't fall into the available options.
When creating interfaces, you get tons of options in the drop down, GPON, CFP2, SFP, Copper, Serial, etc.
But cables don't seem to have as many options.

This specific feature request I am requesting a cable type 'Other' in the drop down list so at least if your cable doesn't fit into the existing list you have some option rather than just leaving it blank.
I'd also like to request cable type "Stacking" if possible. If not, then 'Other' will at least get us somewhat there.

### Use case

When connecting interfaces together, sometimes the cable you're using doesn't fit the currently available options. It would be nice to have an option for 'Other' in those use cases.

### Database changes

None.

### External dependencies

None.

**Comment by cyr0nk0r** [comment]
because there is a difference between 'false' and 'null'.
Leaving the cable type blank could indicate someone forgot to specify the cable type.
Setting the value to type other shows anyone who looks at the value that it was specifically set. It just so happens to be some kind of cable that isn't available within the choices.

**Comment by jeremystretch** [comment]
> It just so happens to be some kind of cable that isn't available within the choices.

But if you know what the type is, and it doesn't exist in NetBox, why not ask that the type be added?

> You could ask your same question for any other drop down field within NetBox that has some kind of 'other' catch all.

Those fields are mandatory; leaving it blank is not an option, so "other" was added as a catch-all.

**Comment by cyr0nk0r** [comment]
Well, I sort of did. I asked for 'Stacking' as an option. Since the cable in question is a proprietary stacking cable for Aruba switches. But it would still be good to have 'other' so someone can at least continue in their NetBox deployment without having to wait however many months it might take to get a feature request pushed through to add cable type 'xyz' into the product.

**Comment by jeremystretch** [comment]
> But it would still be good to have 'other' so someone can at least continue in their NetBox deployment without having to wait however many months it might take to get a feature request pushed through to add cable type 'xyz' into the product.

Again, the `type` field is optional, so this doesn't block anything.

**Comment by cyr0nk0r** [comment]
I mean, I'm not sure what else to say. Does every user have to use NetBox the way you think it should be used?
I submitted the request, it sounds like you don't agree/understand it's purpose. That's fine. I'm not sure what else I can do.

**Comment by jeremystretch** [comment]
> Does every user have to use NetBox the way you think it should be used?

Of course not, and that's the beauty of open source: You always have the freedom to modify it to fit your needs---and to be responsible for those changes indefinitely. For my part, I develop NetBox in accordance with what I think are the best choices to ensure its long-term viability, which unfortunately aren't always appreciated by end users with very little reason to consider use cases beyond their own.

> This specific request is relatively low effort

If effort was a driving factor in whether or not to make a change, I would have given up on this project years ago.

**Comment by cyr0nk0r** [comment]
My comment about low effort is not attempting to disparage your work. I meant, since it's a pretty straight forward change a layman like myself could likely figure out how to quickly just add 1 more option to a drop down list if I could be pointed to the right area where the code for this resides.

**Comment by kkthxbye-code** [comment]
> if you point me to the file where these options are stored I'll work on contributing the code changes.

> if I could be pointed to the right area where the code for this resides.

https://github.com/netbox-community/netbox/blob/301ebe0da3145d822ba1b96c3cbcce0ad59dec89/netbox/dcim/choices.py#L1221

But as Jeremy said you will have to maintain your own fork unless this issue gets accepted.

**Comment by cyr0nk0r** [comment]
I guess I'm just waiting on this FR to be approved then.

**Comment by DanSheps** [comment]
For Aruba, it is a type of DAC, using a proprietary form factor (I think), so I would say for the cable type, you choose "DAC", this is what I would do for Cisco for something like a stackwise stack as well.

**Comment by cyr0nk0r** [comment]
So, still waiting for this to be accepted.

**Comment by cyr0nk0r** [comment]
There is a reason. It's what I outlined above in the feature request.
Just because you don't agree with it doesn't mean there isn't one.
